actions/checkForAndPlaceOrder.ts

```diff
@@ -415,15 +415,58 @@ function _handleGetTradableOrderCall(
   if (error.code === Logger.errors.CALL_EXCEPTION) {
     const errorMessagePrefix =
       "[getTradeableOrderWithSignature] Call Exception";
-    switch (error.errorName) {
+
+    // TODO: When Nethermind hopefully uses the same error format as other nodes, we can remove this hack
+    //
+    // Nethermind does some weird stuff with the error message, so we need to do some magic 💫
+    // Assumptions:
+    // - `error.data` exists for all tested RPC nodes.
+    // - All revert reasons in the code base are expected to return a non-zero `error.data`
+    // - Nethermind, irrespective of the revert reason, returns a zero-bytes `error.data`
+    // Therefore:
+    // - Nethermind: `error.data` in a revert case is `0x` (empty string), with the revert reason buried in
+    //               `error.error.error.data`
+    // - Other nodes: `error.data` in a revert case we expect the revert reason / custom error selector
+    // Therefore, if it's a Nethermind reversion, it takes the format `Reverted 0x01234567` where `01234567` is
+    // the error selector. We can therefore check if the `error.data` is `0x` and if so, we can extract the
+    // error selector from `error.error.error.data` and use that to determine the error.
+    let parsedError = "";
+    if (error.data === "0x") {
+      // Verify that `error.error.error.data` is defined and is a string, otherwise it
+      // could be a different error
+      // TODO: Is there a better way to do this?
+      if (
+        error.error &&
+        error.error.error &&
+        typeof error.error.error.data === "string"
+      ) {
+        // 9 skips the `Reverted ` prefix, and 19 is the end of the error selector
+        parsedError = String(error.error.error.data).slice(9, 19);
+      }
+    } else {
+      parsedError = error.errorName; // If it's not nethermind, we can use the error name
+    }
```

Comment:
should we the hack under a private function so we don't need see the dark magic that retrieves the error?